src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncHelpers.CoreCLR.cs

```diff
@@ -775,10 +781,19 @@ private static ValueTask ValueTaskFromException(Exception ex)
             return new ValueTask<T?>(TaskFromException<T>(ex));
         }
 
+        // Called when capturing execution context for suspension.
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static ExecutionContext? CaptureExecutionContext()
         {
-            return Thread.CurrentThreadAssumedInitialized._executionContext;
+            ExecutionContext? executionContext = Thread.CurrentThreadAssumedInitialized._executionContext;
+            if (executionContext?.m_isFlowSuppressed == true)
+            {
+                // 'null' is the internal representation of default execution
+                // context that contains no variables.
+                executionContext = null;
+            }
```

Comment:
`ExecutionContext.Capture` does something else, returning `ExecutionContext.Default` when the current execution context is null. Do we need to respect that?